pom.xml

```diff
@@ -233,7 +233,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
     <checkstyle.version>9.3</checkstyle.version>
     <surefire.fork.timeout>1200</surefire.fork.timeout>
     <aws-java-sdk.version>1.12.261</aws-java-sdk.version>
-    <frontend-maven-plugin.version>1.12.0</frontend-maven-plugin.version>
+    <frontend-maven-plugin.version>1.12.1</frontend-maven-plugin.version>
```

Comment:
Was this the culprit where it fails to create dir?

Reply:
Oh looks like the build is failing in CI:

https://github.com/apache/ozone/actions/runs/3677105755/jobs/6218636117
```
[INFO] --- frontend-maven-plugin:1.12.1:npx (set pnpm@7.18.1 store path) @ ozone-recon ---
[INFO] npm not inheriting proxy config from Maven
[INFO] Running 'npx pnpm@7.18.1 config set store-dir ~/.pnpm-store' in /home/runner/work/ozone/ozone/hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web
[INFO] npm ERR! code ENOENT
[INFO] npm ERR! syscall lstat
[INFO] npm ERR! path /home/runner/work/ozone/ozone/hadoop-ozone/recon/target/lib
[INFO] npm ERR! errno -2
[INFO] npm ERR! enoent ENOENT: no such file or directory, lstat '/home/runner/work/ozone/ozone/hadoop-ozone/recon/target/lib'
[INFO] npm ERR! enoent This is related to npm not being able to find a file.
[INFO] npm ERR! enoent 
[INFO] 
[INFO] npm ERR! A complete log of this run can be found in:
[INFO] npm ERR!     /home/runner/.npm/_logs/2022-12-12T15_16_06_248Z-debug-0.log
```

Reply:
Yes Same Issue is happening in Local also. Not able to understand root cause

Reply:
@smitajoshi12 - pls check if below link will help. Few community members have reported since v7 of pnpm
https://github.com/pnpm/pnpm/issues/4658

cc: @smengcl

Reply:
We can see bin is created using this command pnpm bin and pnpm store path /home/root1/.pnpm-store/v3 If  I do pnpm run build then local pnpm build is successful but if we do build with Maven then it is failing because of lib folder in target of Recon.

Reply:
I attempted to create directory by adding a plugin in `initialize` phase in recon pom but it didn't seem to work:

```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <executions>
          <execution>
            <id>Workaround: Create lib directory under target</id>
            <phase>initialize</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <target name="Create lib dir">
                <mkdir dir="${project.build.directory}/lib"/>
              </target>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

Reply:
I suspect that `npx` somehow lost its `node_modules` root directory path in the process. Might be a bug in `frontend-maven-plugin` or `npx` or `npm` or `pnpm`. I don't know too much about those nodejs build system so I can't be sure without digging even further.

If this is the case merely creating the empty directory won't do much either.

Need to debug this somehow. Or search for solutions.